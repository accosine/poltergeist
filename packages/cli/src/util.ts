import {
  existsSync,
  realpathSync,
  readFileSync,
  writeFileSync,
} from 'fs-extra';
import { resolve } from 'path';
import * as spawn from 'cross-spawn';
import { prompt } from 'enquirer';

import * as admin from 'firebase-admin';
import * as client from 'firebase-tools';
import * as ensureDefaultCredentials from 'firebase-tools/lib/ensureDefaultCredentials';
import * as logger from 'firebase-tools/lib/logger';
import * as winston from 'winston';

// enable console logging for firebase-tools
logger.add(winston.transports.Console, {
  level: process.env.DEBUG ? 'debug' : 'info',
  showLevel: false,
  colorize: true,
});

export const npmInstall = (cwd: string) =>
  new Promise((resolve, reject) => {
    const npm = spawn('npm', ['--prefix', cwd, 'install'], {
      stdio: 'inherit',
    });
    npm.on('close', resolve);
    npm.on('error', reject);
  });

export const appDirectory = realpathSync(process.cwd());
export const resolveApp = (relativePath: string) =>
  resolve(appDirectory, relativePath);
export const resolveOwn = (relativePath: string) =>
  resolve(__dirname, '../..', relativePath);

export const checkFile = (
  filename: string,
  shouldLog: boolean = true
): boolean => {
  const exists = existsSync(resolveApp(filename));
  if (!exists && shouldLog) {
    console.log(`${filename} missing 🤦🏽‍♀️`);
  }
  return exists;
};

export const getProjectIds = (): string[] =>
  Object.values(
    JSON.parse(readFileSync(resolveApp('.firebaserc'), 'utf8')).projects
  );

export const getProjectId = async () => {
  const projectIds = getProjectIds();
  let projectId;
  if (projectIds.length > 1) {
    projectId = (await prompt<{ projectId: string }>({
      type: 'select',
      name: 'projectId',
      message: 'Which project?',
      choices: projectIds,
    })).projectId;
  } else {
    projectId = projectIds[0];
  }
  try {
    client.use(projectId, {});
  } catch (error) {
    if (
      error.message.startsWith('Command requires authentication, please run')
    ) {
      await initializeAndEnsureAuth(projectId);
      client.use(projectId, {});
    } else {
      throw error;
    }
  }
  return projectId;
};

const initializeApp = (projectId: string) => {
  try {
    admin.initializeApp({ projectId });
  } catch (error) {
    if (error.code !== 'app/duplicate-app') {
      throw error;
    }
  }
};

export const writeEnvFile = (projectId: string, apiKey: string) =>
  writeFileSync(
    resolveApp('.env'),
    `REACT_APP_FIREBASE_PROJECT_ID=${projectId}
REACT_APP_FIREBASE_API_KEY=${apiKey}
REACT_APP_FIREBASE_AUTH_DOMAIN=${projectId}.firebaseapp.com
REACT_APP_FIREBASE_DATABASE_URL=https://${projectId}.firebaseio.com
REACT_APP_FIREBASE_STORAGE_BUCKET=${projectId}.appspot.com`
  );

export const initializeAndEnsureAuth = async (projectId: string) => {
  ensureDefaultCredentials();
  try {
    initializeApp(projectId);
  } catch (error) {
    if (error.code === 'app/invalid-credential') {
      client.login({ localhost: false });
      ensureDefaultCredentials();
      initializeApp(projectId);
    } else {
      console.log(error);
    }
  }
};
