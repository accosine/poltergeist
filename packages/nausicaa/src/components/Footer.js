import React, { Fragment } from 'react';
import { styled } from 'styletron-react';
import { oneLine } from 'common-tags';
import { injectStyle } from 'styletron-utils';
import AmpComponent from './AmpComponent';
import withTheme from '../util/withTheme';

const LogoContainer = styled('div', {
  display: 'flex',
  flexDirection: 'column',
});

const Logo = styled('div', {
  margin: '10vw auto 0 auto',
  width: '26vw',
  padding: '2vw 0 2vw 1vw',
  '@media screen and (min-width: 1024px)': {
    width: '10vw',
    height: 'inherit',
  },
});

const LogoText = styled('svg', {
  margin: '0 auto',
  height: '26vw',
  padding: '0 0 2vw 1vw',
  width: '54vw',
  display: 'block',
  '@media screen and (min-width: 1024px)': {
    width: '22vw',
    height: 'inherit',
  },
});

const Ul = styled('ul', {
  display: 'flex',
  flexDirection: 'column',
  flexWrap: 'wrap',
  height: '40vw',
  listStyleType: 'none',
  padding: '5vw',
  '@media screen and (min-width: 1024px)': {
    height: 'inherit',
    flexDirection: 'row',
    width: '50vw',
    justifyContent: 'space-between',
    padding: '5vw 0 5vw 0',
  },
});

const Li = styled('li', {
  paddingBottom: '3vw',
});

const A = withTheme(
  styled('a', ({ styleProps: { theme } }) => ({
    textDecoration: 'none',
    color: theme.mausgrau,
  }))
);

const AmpUserNotification = withTheme(
  styled(
    AmpComponent('amp-user-notification'),
    ({ styleProps: { theme } }) => ({
      backgroundColor: theme.nausikamint,
      textAlign: 'center',
      padding: '1vw',
    })
  )
);

export default withTheme(({ config, styletron, styleProps }) => {
  const buttonClasses = injectStyle(styletron, {
    padding: '1vw',
    marginTop: '1vw',
    border: 0,
    backgroundColor: styleProps.theme.weißtransparent,
  });

  return (
    <Fragment>
      <footer>
        <LogoContainer>
          <a href="#main">
            <Logo>
              <amp-img
                width={'3'}
                height={'3'}
                src={`${config.media}${config.logo.src}${config.mediasuffix}`}
                alt={config.logo.alt}
                attribution="All Rights Reserved"
                layout="responsive"
              />
            </Logo>
          </a>
          <a href="#main">
            <LogoText>
              <use xlinkHref="#nausika--logotext-use" />
            </LogoText>
          </a>
        </LogoContainer>
        <Ul>
          {/* TODO */}
          {/* <Li>Newsletter</Li> */}
          {/* <Li>Über uns</Li> */}
          {/* <Li>FAQ</Li> */}
          {/* <Li>Werbung</Li> */}
          <Li>
            <A href="/impressum">Impressum</A>
          </Li>
          <Li>
            <A href="/datenschutz">Datenschutz</A>
          </Li>
          <Li>
            <A href="/agb">AGB</A>
          </Li>
          <Li>
            <A href="/rss">RSS</A>
          </Li>
          <Li>
            <A href={`https://www.facebook.com/${config.vanityurl}`}>
              Facebook
            </A>
          </Li>
          <Li>
            <A href={`https://www.twitter.com/${config.vanityurl}`}>Twitter</A>
          </Li>
        </Ul>
      </footer>
      <AmpUserNotification
        layout="nodisplay"
        id="amp-user-notification1"
        dangerouslySetInnerHTML={{
          __html: oneLine`
            <p>
              nausika nutzt Cookies in deinem Browser. Wir speichern darin allerlei
              Daten - aber keine Sorge, nichts Wildes. Mehr langweiliges bla bla
              über Cookies kannst du <a href="/datenschutz/">hier lesen...</a>
            </p>
            <button class="${buttonClasses}" on="tap:amp-user-notification1.dismiss">
              Cookies finde ich ganz ok
            </button>
            `,
        }}
      />
    </Fragment>
  );
});
