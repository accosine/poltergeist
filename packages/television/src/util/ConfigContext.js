import React, { createContext } from 'react';

const { Provider, Consumer } = createContext({});

const withConfig = Component => props => (
  <Consumer>{data => <Component {...props} config={data} />}</Consumer>
);

export { Consumer as ConfigConsumer, Provider as ConfigProvider, withConfig };
