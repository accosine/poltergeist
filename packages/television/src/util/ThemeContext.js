import React, { createContext } from 'react';

const { Provider, Consumer } = createContext({});

const withTheme = Component => ({ context, ...props }) => (
  <Consumer>
    {data => <Component {...props} $context={context} $theme={data} />}
  </Consumer>
);

export { Consumer as ThemeConsumer, Provider as ThemeProvider, withTheme };
