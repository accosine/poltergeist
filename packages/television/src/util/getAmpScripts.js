export default (usedShortcodes, plugins) => {
  const pluginScripts = plugins
    .filter(
      plugin =>
        plugin.shortcode &&
        plugin.shortcode.theme &&
        plugin.shortcode.theme.ampScript
    )
    .reduce(
      (acc, plugin) => ({
        [plugin.shortcode.label]: plugin.shortcode.theme.ampScript,
      }),
      {}
    );

  const scripts = {
    carousel: 'carousel',
    youtube: 'youtube',
    gif: 'anim',
    soundcloud: 'soundcloud',
    vimeo: 'vimeo',
    instagram: 'instagram',
    gfycat: 'gfycat',
    fittext: 'fit-text',
    facebook: 'facebook',
    facebookcomments: 'facebook-comments',
    pinterest: 'pinterest',
    playbuzz: 'playbuzz',
    twitter: 'twitter',
    iframe: 'iframe',
    ...pluginScripts,
  };

  return usedShortcodes
    .map(shortcode => (scripts[shortcode] ? scripts[shortcode] : false))
    .filter(Boolean);
};
