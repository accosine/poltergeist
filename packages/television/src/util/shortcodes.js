import Shortcode from './shortcode-parser';
import { oneLine } from 'common-tags';
import addSizeSuffix from './addSizeSuffix';
// import { injectStyle } from 'styletron-utils';

export default (config, plugins) => {
  const storageurl = config.media;
  const storagesuffix = config.mediasuffix;

  // TODO: make srcset sizes configurable
  const shortcodes = {
    image: (str, params, { styletron }) => {
      const figText = `${params.caption || ''}${
        params.attribution ? ' (' + params.attribution + ')' : ''
      }`;
      const figOpen = params.caption || params.attribution ? '<figure>' : '';
      const figClose =
        params.caption || params.attribution
          ? `<figcaption>${figText}</figcaption></figure>`
          : '';
      const lightbox = params.lightbox
        ? 'on="tap:lightbox1" role="button" tabindex="0"'
        : '';
      return `<div>${figOpen}<amp-img
                  ${
                    params.fill
                      ? ''
                      : `width=${params.width || 1}
                  height=${params.height || 1}`
                  }
                  src="${storageurl}${addSizeSuffix(
        params.name,
        '-s'
      )}${storagesuffix}"
                  srcset="${storageurl}${addSizeSuffix(
        params.name,
        '-l'
      )}${storagesuffix} 1280w,
                  ${storageurl}${addSizeSuffix(
        params.name,
        '-m'
      )}${storagesuffix} 640w,
                  ${storageurl}${addSizeSuffix(
        params.name,
        '-s'
      )}${storagesuffix} 320w"
                  alt="${params.alttext || ''}"
                  attribution="${params.attribution || ''}" ${lightbox}
                  layout="${
                    params.fill ? 'fill' : 'responsive'
                  }"></amp-img/>${figClose}</div>`;
    },
    // video: (str, params) =>
    //   `<div><amp-video
    //     layout="responsive"
    //     width=${params.width || 16}
    //     height=${params.height || 9}
    //     ${params.autoplay ? 'autoplay' : ''}
    //     ${params.loop ? 'loop' : ''}
    //      src="${params.url || ''}"
    //      poster="${params.poster || ''}">
    //        <div fallback>
    //          <p>Your browser doesn’t support HTML5 video</p>
    //        </div>
    //   </amp-video></div>`,
    // gif: (str, params) =>
    //   `<div><amp-anim
    //     layout="responsive"
    //     width=${params.width || 1}
    //     height=${params.height || 1}
    //     src="/${config.media}/${params.name}">
    //     <amp-img
    //     placeholder
    //     layout="responsive"
    //     width=${params.width || 1}
    //     height=${params.height || 1}
    //     src="/${config.media}/placeholder-${params.name}">
    //     </amp-img>
    //   </amp-anim></div>`,
    // audio: (str, params) =>
    //   `<div><amp-audio
    //     src="${params.url}">
    //     <div fallback>
    //     <p>Your browser doesn’t support HTML5 audio</p>
    //     </div>
    //     <source type="audio/mpeg" src="foo.mp3">
    //     <source type="audio/ogg" src="foo.ogg">
    //   </amp-audio></div>`,
    soundcloud: (str, params) =>
      oneLine`<div><amp-soundcloud
          height=${params.height || 166}
          layout="fixed-height" ${
            params.color && !params.visual ? `data-color="${params.color}"` : ''
          } data-trackid="${params.id}"
          ${params.visual ? 'data-visual="true"' : ''}>
        </amp-soundcloud></div>`,
    carousel: (str, { autoplay, delay, controls, width, height, loop }) =>
      oneLine`<div><amp-carousel
                type="slides"
                layout="responsive"
                ${autoplay ? `autoplay delay="${delay || 3000}"` : ''}
                ${controls ? 'controls' : ''}
                ${loop ? 'loop' : ''}
                width=${width || 4}
                height=${height || 3}>
                ${str}
              </amp-carousel></div>`,
    facebook: (str, { url, width, height, video }) =>
      oneLine`<div><amp-facebook width="${width}" height="${height}"
                layout="responsive"
                ${video ? 'data-embed-as="video"' : ''}
                data-href="${url}">
              </amp-facebook></div>`,
    facebookcomments: (str, { url, width, height, numposts }) =>
      oneLine`<div><amp-facebook-comments
                width=${width || 1}
                height=${height || 1}
                layout="responsive"
                data-numposts="${numposts || 5}"
                data-href="${url}">
              </amp-facebook-comments></div>`,
    fittext: (str, { width, height, min, max }) =>
      oneLine`<div><amp-fit-text
                width=${width || 4}
                height=${height || 3}
                min-font-size=${min || 10}
                max-font-size=${max || 52}
                layout="responsive">
              ${str}
            </amp-fit-text></div>`,
    gfycat: (str, params) =>
      oneLine`<div><amp-gfycat
                data-gfyid="${params.id}"
                width="${params.width || 4}"
                height="${params.height || 3}"
                layout="responsive">
                ${params.noautoplay ? 'noautoplay' : ''}>
              </amp-gfycat></div>`,
    iframe: (str, params) =>
      oneLine`<div><amp-iframe
                sandbox="allow-scripts allow-same-origin allow-popups"
                src=${params.url || ''}
                width=${params.width || 1}
                height=${params.height || 1}
                layout="responsive">
              </amp-iframe></div>`,
    instagram: (str, params) =>
      oneLine`<div><amp-instagram
                data-shortcode="${params.id || ''}"
                ${params.captioned ? 'data-captioned' : ''}
                width=${params.width || 1}
                height=${params.height || 1}
                layout="responsive">
              </amp-instagram></div>`,
    pinterest: (str, { width, height, url }) =>
      oneLine`<div><amp-pinterest
                width=${width || '236'}
                height=${height || '326'}
                data-do="embedPin"
                data-url="${url}">
              </amp-pinterest></div>`,
    playbuzz: (str, { height, url, iteminfo }) =>
      oneLine`<div><amp-playbuzz
                src="${url}"
                height=${height}
                ${iteminfo ? 'data-item-info="true"' : ''}>
              </amp-playbuzz></div>`,
    twitter: (str, { id, width, height, cardshidden }) =>
      oneLine`<div><amp-twitter
                width=${width}
                height=${height}
                layout="responsive"
                data-align="center"
                data-tweetid=${id}
                ${cardshidden ? 'data-cards="hidden"' : ''}>
              </amp-twitter></div>`,
    vine: (str, params) =>
      oneLine`<div><amp-vine
                data-vineid="${params.vineid || ''}"
                width=${params.width || 400}
                height=${params.height || 400}>
                layout="responsive">
              </amp-vine></div>`,
    vimeo: (str, params) =>
      oneLine`<div><amp-vimeo
                data-videoid="${params.videoid || ''}"
                width=${params.width || 500}
                height=${params.height || 281}
                layout="responsive">
              </amp-vimeo></div>`,
    youtube: (str, params) => {
      return oneLine`<div><amp-youtube
                data-videoid="${params.videoid || ''}"
                width=${params.width || 480}
                height=${params.height || 270}
                layout="responsive">
              </amp-youtube></div>`;
    },
    // EXAMPLE: styling with classes
    // youtube: (str, params, { styletron }) => {
    //   const classes = injectStyle(styletron, {
    //     border: '1px solid red',
    //     borderRadius: '100%',
    //   });
    //   return oneLine`<div><amp-youtube class="${classes}"
    //             data-videoid="${params.videoid || ''}"
    //             width=${params.width || 480}
    //             height=${params.height || 270}
    //             layout="responsive">
    //           </amp-youtube></div>`;
    // },
  };

  plugins
    .filter(plugin => plugin.shortcode && plugin.shortcode.theme)
    .forEach(plugin =>
      Shortcode.add(plugin.shortcode.label, plugin.shortcode.theme.transform)
    );

  for (let s in shortcodes) {
    Shortcode.add(s, shortcodes[s]);
  }

  return (text, styletron) => Shortcode.parse(text, { styletron, config });
};
