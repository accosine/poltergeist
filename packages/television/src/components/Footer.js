import React, { Fragment } from 'react';
import { styled } from 'styletron-react';
import { oneLine } from 'common-tags';
import AmpComponent from './AmpComponent';
import Link from './Link';
import { withTheme } from '../util/ThemeContext';

const LogoContainer = styled('div', {
  display: 'flex',
  flexDirection: 'column',
});

const LogoText = withTheme(
  styled('svg', ({ $theme }) => ({
    fill: $theme.grey[900],
    margin: '0 auto',
    height: '26vw',
    padding: '0 0 2vw 1vw',
    width: '54vw',
    display: 'block',
    '@media screen and (min-width: 1024px)': {
      width: '22vw',
      height: 'inherit',
    },
  }))
);

const Ul = styled('ul', {
  display: 'flex',
  flexDirection: 'column',
  flexWrap: 'wrap',
  height: '40vw',
  listStyleType: 'none',
  padding: '5vw',
  '@media screen and (min-width: 1024px)': {
    height: 'inherit',
    flexDirection: 'row',
    width: '50vw',
    justifyContent: 'space-between',
    padding: '5vw 0 5vw 0',
  },
});

const Li = styled('li', {
  paddingBottom: '3vw',
});

const A = withTheme(
  styled(Link, ({ $theme }) => ({
    textDecoration: 'none',
    color: $theme.grey[700],
  }))
);

const UserConsent = withTheme(
  styled(AmpComponent('amp-user-notification'), ({ $theme }) => ({
    backgroundColor: $theme.grey[200],
    textAlign: 'center',
    padding: '1vw',
  }))
);

const SocialMedia = ({ name, handle, base }) =>
  handle ? (
    <Li>
      <A href={`${base}${handle}`} target="_blank">
        {name}
      </A>
    </Li>
  ) : null;

export default withTheme(({ config, styletron, $theme }) => {
  const buttonClasses = styletron.renderStyle({
    padding: '1vw',
    marginTop: '1vw',
    border: 0,
    backgroundColor: $theme.transparentwhite,
  });

  return (
    <Fragment>
      <footer>
        <LogoContainer>
          <a href="#main">
            <LogoText>
              <use xlinkHref="#biglogo" />
            </LogoText>
          </a>
        </LogoContainer>
        <Ul>
          <Li>
            <A href="/imprint">Imprint</A>
          </Li>
          <Li>
            <A href="/privacypolicy">Privacy</A>
          </Li>
          {config.socialmedia.map(props => (
            <SocialMedia key={props.name} {...props} />
          ))}
        </Ul>
      </footer>
      <UserConsent
        layout="nodisplay"
        id="amp-user-notification1"
        dangerouslySetInnerHTML={{
          __html: oneLine`
            <p>
              Let's have a cookie! They are not only delicious, they help us
              analyzing how you use this website. We use them so store some data
              in them (but not too much). Don't be afraid, we won't give this
              data to anyone (promise ðŸ¤ž).
              If you like to read really boring stuff you can read all about it
              in <a href="/privacypolicy">our privacy policy here...</a>
            </p>
            <button class="${buttonClasses}" on="tap:amp-user-notification1.dismiss">
              Yay cookies, nom nom nom
            </button>
            `,
        }}
      />
    </Fragment>
  );
});
