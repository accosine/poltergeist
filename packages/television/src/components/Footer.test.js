import React from 'react';
import renderer from 'react-test-renderer';
import { Provider as StyletronProvider } from 'styletron-react';
import { Server as StyletronServer } from 'styletron-engine-atomic';
import { ThemeProvider } from '../util/ThemeContext';

import { config } from '../fixtures';
import colors from '../colors.js';
import Footer from './Footer';

jest.mock('styletron-react', () => {
  const React = require.requireActual('react');
  const originalStyletron = require.requireActual('styletron-react');
  return {
    ...originalStyletron,
    styled: (name, stylesCreator) => {
      return props =>
        React.createElement(name, {
          ...Object.keys(props).reduce((acc, k) => {
            if (k[0] === '$') {
              acc[`data-styleprop-${k.substr(1)}`] = props[k];
            } else {
              acc[k] = props[k];
            }
            return acc;
          }, {}),
          'data-style':
            typeof stylesCreator === 'function'
              ? stylesCreator(props)
              : stylesCreator,
        });
    },
  };
});

test('Link changes the class when hovered', () => {
  const styletron = new StyletronServer();
  const component = renderer.create(
    <StyletronProvider value={styletron}>
      <ThemeProvider value={colors}>
        <Footer styletron={styletron} config={config.application} />
      </ThemeProvider>
    </StyletronProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
