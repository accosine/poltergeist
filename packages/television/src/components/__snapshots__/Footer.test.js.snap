// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
Array [
  <footer>
    <div
      data-style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <a
        href="#main"
      >
        <svg
          data-style={
            Object {
              "@media screen and (min-width: 1024px)": Object {
                "height": "inherit",
                "width": "22vw",
              },
              "display": "block",
              "fill": "#333333",
              "height": "26vw",
              "margin": "0 auto",
              "padding": "0 0 2vw 1vw",
              "width": "54vw",
            }
          }
          data-styleprop-theme={
            Object {
              "black": "#000000",
              "collection": [Function],
              "email": "rgb(211, 76, 66)",
              "facebook": "#3C3CDE",
              "github": "#24292e",
              "gradientrotation": "135deg",
              "grey": Object {
                "100": "#F9FAFB",
                "200": "#f1f1f1",
                "400": "#C1C1C1",
                "500": "#969696",
                "700": "#585252",
                "900": "#333333",
              },
              "instagram": "url(#instagram-gradient)",
              "primary": "#FF6500",
              "snapchat": "#fffc00",
              "transparent": "rgba(255, 255, 255, 0)",
              "transparentwhite": "rgba(255,255,255,0.64)",
              "twitter": "#8FD6EC",
              "whatsapp": "#06C34B",
              "white": "#fff",
            }
          }
        >
          <use
            xlinkHref="#biglogo"
          />
        </svg>
      </a>
    </div>
    <ul
      data-style={
        Object {
          "@media screen and (min-width: 1024px)": Object {
            "flexDirection": "row",
            "height": "inherit",
            "justifyContent": "space-between",
            "padding": "5vw 0 5vw 0",
            "width": "50vw",
          },
          "display": "flex",
          "flexDirection": "column",
          "flexWrap": "wrap",
          "height": "40vw",
          "listStyleType": "none",
          "padding": "5vw",
        }
      }
    >
      <li
        data-style={
          Object {
            "paddingBottom": "3vw",
          }
        }
      >
        <a
          data-style={
            Object {
              "color": "#585252",
              "textDecoration": "none",
            }
          }
          data-styleprop-theme={
            Object {
              "black": "#000000",
              "collection": [Function],
              "email": "rgb(211, 76, 66)",
              "facebook": "#3C3CDE",
              "github": "#24292e",
              "gradientrotation": "135deg",
              "grey": Object {
                "100": "#F9FAFB",
                "200": "#f1f1f1",
                "400": "#C1C1C1",
                "500": "#969696",
                "700": "#585252",
                "900": "#333333",
              },
              "instagram": "url(#instagram-gradient)",
              "primary": "#FF6500",
              "snapchat": "#fffc00",
              "transparent": "rgba(255, 255, 255, 0)",
              "transparentwhite": "rgba(255,255,255,0.64)",
              "twitter": "#8FD6EC",
              "whatsapp": "#06C34B",
              "white": "#fff",
            }
          }
          href="/imprint"
        >
          Imprint
        </a>
      </li>
      <li
        data-style={
          Object {
            "paddingBottom": "3vw",
          }
        }
      >
        <a
          data-style={
            Object {
              "color": "#585252",
              "textDecoration": "none",
            }
          }
          data-styleprop-theme={
            Object {
              "black": "#000000",
              "collection": [Function],
              "email": "rgb(211, 76, 66)",
              "facebook": "#3C3CDE",
              "github": "#24292e",
              "gradientrotation": "135deg",
              "grey": Object {
                "100": "#F9FAFB",
                "200": "#f1f1f1",
                "400": "#C1C1C1",
                "500": "#969696",
                "700": "#585252",
                "900": "#333333",
              },
              "instagram": "url(#instagram-gradient)",
              "primary": "#FF6500",
              "snapchat": "#fffc00",
              "transparent": "rgba(255, 255, 255, 0)",
              "transparentwhite": "rgba(255,255,255,0.64)",
              "twitter": "#8FD6EC",
              "whatsapp": "#06C34B",
              "white": "#fff",
            }
          }
          href="/privacypolicy"
        >
          Privacy
        </a>
      </li>
      <li
        data-style={
          Object {
            "paddingBottom": "3vw",
          }
        }
      >
        <a
          data-style={
            Object {
              "color": "#585252",
              "textDecoration": "none",
            }
          }
          data-styleprop-theme={
            Object {
              "black": "#000000",
              "collection": [Function],
              "email": "rgb(211, 76, 66)",
              "facebook": "#3C3CDE",
              "github": "#24292e",
              "gradientrotation": "135deg",
              "grey": Object {
                "100": "#F9FAFB",
                "200": "#f1f1f1",
                "400": "#C1C1C1",
                "500": "#969696",
                "700": "#585252",
                "900": "#333333",
              },
              "instagram": "url(#instagram-gradient)",
              "primary": "#FF6500",
              "snapchat": "#fffc00",
              "transparent": "rgba(255, 255, 255, 0)",
              "transparentwhite": "rgba(255,255,255,0.64)",
              "twitter": "#8FD6EC",
              "whatsapp": "#06C34B",
              "white": "#fff",
            }
          }
          href="mailto:yo@example.com"
          target="_blank"
        >
          Email
        </a>
      </li>
    </ul>
  </footer>,
  <amp-user-notification
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p> Let's have a cookie! They are not only delicious, they help us analyzing how you use this website. We use them so store some data in them (but not too much). Don't be afraid, we won't give this data to anyone (promise ðŸ¤ž). If you like to read really boring stuff you can read all about it in <a href=\\"/privacypolicy\\">our privacy policy here...</a> </p> <button class=\\"ae af ag ah\\" on=\\"tap:amp-user-notification1.dismiss\\"> Yay cookies, nom nom nom </button>",
      }
    }
    data-style={
      Object {
        "backgroundColor": "#f1f1f1",
        "padding": "1vw",
        "textAlign": "center",
      }
    }
    data-styleprop-theme={
      Object {
        "black": "#000000",
        "collection": [Function],
        "email": "rgb(211, 76, 66)",
        "facebook": "#3C3CDE",
        "github": "#24292e",
        "gradientrotation": "135deg",
        "grey": Object {
          "100": "#F9FAFB",
          "200": "#f1f1f1",
          "400": "#C1C1C1",
          "500": "#969696",
          "700": "#585252",
          "900": "#333333",
        },
        "instagram": "url(#instagram-gradient)",
        "primary": "#FF6500",
        "snapchat": "#fffc00",
        "transparent": "rgba(255, 255, 255, 0)",
        "transparentwhite": "rgba(255,255,255,0.64)",
        "twitter": "#8FD6EC",
        "whatsapp": "#06C34B",
        "white": "#fff",
      }
    }
    id="amp-user-notification1"
    layout="nodisplay"
  />,
]
`;
